### PR TITLE
Search backend: Rename `commit.CommitSearch` to `commit.CommitSearchJob`

### PR DESCRIPTION
**Pull request by tbliu98**
Related to #34009

Because `commit.CommitSearch` implements `job.Job`, renaming `commit.CommitSearch` to `commit.CommitSearchJob` for clarity when reading the code and looking at traces.

## Test plan
Semantics-preserving.

### CODE REVIEW

File: internal/search/job/jobutil/job.go [hunk 1]
```diff
@@ -762,7 +762,7 @@ func optimizeJobs(baseJob job.Job, inputs *run.SearchInputs, q query.Basic) (job
 				}
 				return currentJob
 
-			case *commit.CommitSearch:
+			case *commit.CommitSearchJob:
 				if exists("Commit") || exists("Diff") {
```
Comment (camdencheek) [inline internal/search/job/jobutil/job.go:766]: We'll need to update any references to the name string as well

```suggestion
				if exists("CommitSearchJob") || exists("DiffSearchJob") {
```

(sorry for the using strings as identification here. the stuff we worked on today should  make this unnecessary)

**Review by camdencheek — APPROVED** [review]
I would expect some changes to the tests for pretty-printing...do we not have any tests that create job trees with commit/diff jobs?

File: internal/search/job/jobutil/job.go (hunk 1 above)
Comment (tbliu98) [reply internal/search/job/jobutil/job.go]: Ah, thanks. Fixed

**Comment by tbliu98** [comment]
> I would expect some changes to the tests for pretty-printing...do we not have any tests that create job trees with commit/diff jobs?

We do, I forgot about the autogold files until the build complained. Just fixed those.

**Pull request merged**